solr/solrj/src/java/org/apache/solr/common/cloud/SolrZkClient.java

```diff
@@ -83,6 +87,46 @@ public class SolrZkClient implements Closeable {
 
   private ZkCmdExecutor zkCmdExecutor;
 
+  private final ZkMetrics metrics = new ZkMetrics();
+
+  public static class ZkMetrics implements ReflectMapWriter {
+    @JsonProperty
+    public final LongAdder watchesFired = new LongAdder();
```

Comment:
I was bit thinking about number of watches from solr-node. Should be increment watch when installed and decrement when it trigger?

Reply:
yeah, this could be another metric that we collect